Okta/OktaKeychain.swift

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2017, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+enum OktaKeychainError: Error {
+    case codingError
+    case failed(String)
+}
+
+class OktaKeychain: NSObject {
+
+    /**
+     Stores an item securely in the Keychain.
+     - parameters:
+     - key: Hash to reference the stored Keychain item
+     - string: String to store inside of the keychain
+     */
+    class func set(key: String, string: String, expiration: TimeInterval? = nil, accessGroup: String? = nil, accessibility: CFString? = nil) throws {
+        guard let objectData = string.data(using: .utf8) else {
+            throw OktaKeychainError.codingError
+        }
+        try set(key: key, data: objectData, expiration: expiration, accessGroup: accessGroup, accessibility: accessibility)
+    }
+    
+    /**
+     Stores an item securely in the Keychain.
+     - parameters:
+     - key: Hash to reference the stored Keychain item
+     - data: Data to store inside of the keychain
+     */
+    class func set(key: String, data: Data, expiration: TimeInterval? = nil, accessGroup: String? = nil, accessibility: CFString? = nil) throws {
+        var q = [
+            kSecClass as String: kSecClassGenericPassword as String,
+            kSecValueData as String: data,
+            kSecAttrAccount as String: key,
+            kSecAttrAccessible as String: accessibility ?? kSecAttrAccessibleWhenUnlockedThisDeviceOnly
+        ] as [String : Any]
+        
+        if let accessGroup = accessGroup {
+            q[kSecAttrAccessGroup as String] = accessGroup
+        }
+        
+        let cfDictionary = q as CFDictionary
+        // Delete existing (if applicable)
+        SecItemDelete(cfDictionary)
+        
+        let sanityCheck = SecItemAdd(cfDictionary, nil)
+        if sanityCheck != noErr {
+            throw OktaKeychainError.failed(sanityCheck.description)
+        }
+        
+        if let expiration = expiration {
```

Comment:
We're not using this currently, so lets go ahead and remove the `expiration` argument from our setter.

Reply:
removed